src/WebJobs.Script.WebHost/Filters/AuthorizationLevelAttribute.cs

```diff
@@ -32,8 +32,9 @@ public AuthorizationLevelAttribute(AuthorizationLevel level)
             }
 
             ISecretManager secretManager = actionContext.ControllerContext.Configuration.DependencyResolver.GetService<ISecretManager>();
+            var settings = actionContext.ControllerContext.Configuration.DependencyResolver.GetService<WebHostSettings>();
 
-            if (!await IsAuthorizedAsync(actionContext.Request, Level, secretManager))
+            if (!settings.IsSelfHost && !await IsAuthorizedAsync(actionContext.Request, Level, secretManager))
```

Comment:
You mean to use your new DisableAuth setting here. Also let's rename it to IsAuthDisabled to match the Is pattern of the other bools.